src/browser.js

```diff
@@ -148,35 +204,20 @@ function save(namespaces) {
  */
 
 function load() {
-  try {
-    return exports.storage.debug;
-  } catch(e) {}
-
-  // If debug isn't set in LS, and we're in Electron, try to load $DEBUG
-  if (typeof process !== 'undefined' && 'env' in process) {
-    return process.env.DEBUG;
+  if (exports.storage) {
+    exports.storage.get('debug', exports.enable);
```

Comment:
Sorry, I'll elaborate:

> There's no enabled and there's no exports.names

Not in this file, no. We need to jump over to `debug.js` to understand what happens. In that file there is [a method][1] called `enabled`. It is run from [createDebug][2], i.e. when `debug('foo')` is called. Currently, we're in a situation where `enabled` is very likely to be called before [exports.names][3] is set, which is a precondition of that method. Thus we are breaking the contract of `enabled`.

`exports.names` are set by [enable][4], which is passed as a callback in [load][5]. Problem is, now that `storage.get` is async, the order which the assignment happens is mixed up. With `localStorage` we can assume synchronous behavior and it works, but this is not the case for WebExtensions.

```javascript
import debug from 'debug';
const log = debug('foo');

log('Hello!'); // <-- Never logs because debug was not enabled when `debug('foo')` was called

setTimeout(() => {
  debug('bar')('Hi!'); // <-- (Probably) runs because the promise has resolved and the callback fired
}, 1000);
```

[1]: https://github.com/andreineculau/debug/blob/patch-1-without-ua-sniff/src/debug.js#L176
[2]: https://github.com/andreineculau/debug/blob/patch-1-without-ua-sniff/src/debug.js#L118
[3]: https://github.com/andreineculau/debug/blob/patch-1-without-ua-sniff/src/debug.js#L183
[4]: https://github.com/andreineculau/debug/blob/patch-1-without-ua-sniff/src/debug.js#L153
[5]: https://github.com/andreineculau/debug/blob/patch-1-without-ua-sniff/src/browser.js#L207